### PR TITLE
Add autoconfig support to installer

### LINKED ISSUE #157: feat: use new autoconfigure command
from https://github.com/OpenVoiceOS/ovos-config/pull/158

```
(.venv) [miro@miro-asustufgamingf15fx506hmfx506hm lang_configs]$ ovos-config autoconfigure --lang ca-nw --male --online
Standardized lang-code: ca-NW
Merged config: /home/miro/PycharmProjects/ovos-config/ovos_config/recommends/base/ca-es.conf
Merged config: /home/miro/PycharmProjects/ovos-config/ovos_config/recommends/online_stt/ca-es.conf
Merged config: /home/miro/PycharmProjects/ovos-config/ovos_config/recommends/online_male/ca-nw.conf
2024-09-26 00:21:50.315 - OVOS - ovos_plugin_manager.utils:find_plugins:136 - ERROR - Failed to load plugin entry point EntryPoint(name='ovos-tts-plugin-mimic', value='ovos_tts_plugin_mimic:MimicTTSPlugin', group='mycroft.plugin.tts'): No module named 'ovos_utils.configuration'
Available STT plugins:
  - 'ovos-stt-plugin-vosk'
  - 'ovos-stt-plugin-vosk-streaming'
  - 'ovos-stt-plugin-chromium'
  - 'ovos-stt-plugin-server'
  - 'ovos-stt-plugin-dummy'
Available TTS plugins:
  - 'ovos-tts-plugin-server'
  - 'ovos-tts-plugin-polly'
  - 'ovos-tts-plugin-dummy'
Config updated: /home/miro/.config/mycroft/mycroft.conf
{
  "lang": "ca-NW",
  "tts": {
    "ovos-tts-plugin-server": {
      "host": [
        "https://tts.smartgic.io/matxa",
        "https://matxa.ziggyai.online"
      ],
      "voice": "nord-occidental/pere"
    },
    "module": "ovos-tts-plugin-server",
    "fallback_module": ""
  },
  "system_unit": "metric",
  "time_format": "full",
  "date_format": "DMY",
  "stt": {
    "ovos-stt-plugin-server": {
      "url": [
        "https://stt.smartgic.io/citrinet",
        "https://citrinetstt.ziggyai.online"
      ]
    },
    "module": "ovos-stt-plugin-server",
    "fallback_module": ""
  },
  "spoken_time_format": "half",
  "temperature_unit": "celsius",
  "windspeed_unit": "km/h",
  "precipitation_unit": "mm"
}

```

### PR DESCRIPTION
**Pull request by suvanbanerjee**
Closes #157

## Summary by CodeRabbit

- **New Features**
	- Introduced automatic configuration for Speech-to-Text (STT) and Text-to-Speech (TTS) based on user language settings.
	- Enhanced configuration for STT and TTS auto-configuration when the installation profile is not set to "server".
	- Added performance improvements in task execution for managing virtual environments.

- **Bug Fixes**
	- Improved readability and performance in existing tasks without altering functionality.

### CODE REVIEW

**Comment by suvanbanerjee** [comment]
@goldyfruit will this file structure work, I will ask user for autoconfig questions and then run the command accordingly, to tackle the region issue we can find region using their ip to detect their country so if someone chooses English in Australia then its en-au,
en-> from his language choice
au-> from his country ip

**Comment by JarbasAl** [comment]
> @goldyfruit will this file structure work, I will ask user for autoconfig questions and then run the command accordingly, to tackle the region issue we can find region using their ip to detect their country so if someone chooses English in Australia then its en-au, en-> from his language choice au-> from his country ip

i wouldnt worry about this for now, as the translations currently are all for en-us , setting a different code might just cause issues with no benefit

only portuguese currently differentiates between pt-br and pt-pt, no other language supports different dialect lang codes (yet)

**Comment by goldyfruit** [comment]
I would not prompt any questions about the auto configuration. By default if **French** is selected then the assistant should be configured to in **French**, etc...

About the male or female voice, this question should be handle by the core team as its part of the OVOS's persona.

@JarbasAl  @j1nx Do we want to have a male or female persona?

File: ansible/roles/ovos_installer/tasks/ovos.yml
Comment (goldyfruit) [inline ansible/roles/ovos_installer/tasks/ovos.yml:1]: - There is no need for the `vars` dictionary as you already have the variable defined.
- Use the FQCN for the module.
- `args` is deprecated, executable should be an argument of `shell`.

I would recommend to have a look here: https://github.com/OpenVoiceOS/ovos-installer/blob/5a7e631f7f89eb8872a57dacb149654698df40a9/ansible/roles/ovos_installer/tasks/virtualenv/venv.yml#L103-L118

**Comment by suvanbanerjee** [comment]
> If not prompting user, keep it male

I guess the standard for voice assistant is female ?

**Comment by JarbasAl** [comment]
> > If not prompting user, keep it male
> 
> I guess the standard for voice assistant is female ?

historically we have used the male voice from alan pope, which goes well with "hey mycroft". let's keep it that way

besides, its a good differentiation factor in comparison with other voice assistants

**Comment by goldyfruit** [comment]
Just pushed a commit, please have a look.

**Comment by goldyfruit** [comment]
`mycroft.conf` generated by the `ovos-config` seems off.

```
{
  "log_level": "INFO",
  "play_wav_cmdline": "pw-play %1",
  "play_mp3_cmdline": "pw-play %1",
  "lang": "ca-ES",
  "listener": {
    "remove_silence": true,
    "VAD": {
      "module": "ovos-vad-plugin-silero"
    },
    "instant_listen": true
  },
  "skills": {
    "installer": {
      "allow_pip": true,
      "allow_alphas": true
    }
  },
  "websocket": {
    "max_msg_size": 25
  },
  "PHAL": {
    "ovos-PHAL-plugin-wifi-setup": {
      "module": "ovos-PHAL-plugin-wifi-setup",
      "lang": "ca-ES",
      "secondary_langs": [],
      "system_unit": "metric",
      "temperature_unit": "celsius",
      "windspeed_unit": "km/h",
      "precipitation_unit": "mm",
      "time_format": "full",
      "spoken_time_format": "half",
      "date_format": "DMY",
      "opt_in": false,
      "confirm_listening": true,
      "ready_settings": [
        "skills"
      ],
      "ignore_logs": [
        "enclosure.mouth.viseme",
        "enclosure.mouth.display"
      ],
      "enable_old_audioservice": true,
      "debug": false,
      "log_level": "INFO",
      "play_wav_cmdline": "pw-play %1",
      "play_mp3_cmdline": "pw-play %1",
      "first_boot": false
    }
  },
  "tts": {
    "ovos-tts-plugin-server": {
      "host": [
        "https://tts.smartgic.io/matxa",
        "https://matxa.ziggyai.online"
      ],
      "voice": "central/grau"
    },
    "module": "ovos-tts-plugin-server"
  },
  "stt": {
    "ovos-stt-plugin-server": {
      "url": [
        "https://stt.smartgic.io/citrinet",
        "https://citrinetstt.ziggyai.online"
      ]
    },
    "module": "ovos-stt-plugin-server",
    "fallback_module": ""
  },
  "system_unit": "metric",
  "temperature_unit": "celsius",
  "windspeed_unit": "km/h",
  "precipitation_unit": "mm",
  "time_format": "full",
  "spoken_time_format": "half",
  "date_format": "DMY"
}
```

**Comment by goldyfruit** [comment]
> `mycroft.conf` generated by the `ovos-config` seems off.
> 
> ```
> {
>   "log_level": "INFO",
>   "play_wav_cmdline": "pw-play %1",
>   "play_mp3_cmdline": "pw-play %1",
>   "lang": "ca-ES",
>   "listener": {
>     "remove_silence": true,
>     "VAD": {
>       "module": "ovos-vad-plugin-silero"
>     },
>     "instant_listen": true
>   },
>   "skills": {
>     "installer": {
>       "allow_pip": true,
>       "allow_alphas": true
>     }
>   },
>   "websocket": {
>     "max_msg_size": 25
>   },
>   "PHAL": {
>     "ovos-PHAL-plugin-wifi-setup": {
>       "module": "ovos-PHAL-plugin-wifi-setup",
>       "lang": "ca-ES",
>       "secondary_langs": [],
>       "system_unit": "metric",
>       "temperature_unit": "celsius",
>       "windspeed_unit": "km/h",
>       "precipitation_unit": "mm",
>       "time_format": "full",
>       "spoken_time_format": "half",
>       "date_format": "DMY",
>       "opt_in": false,
>       "confirm_listening": true,
>       "ready_settings": [
>         "skills"
>       ],
>       "ignore_logs": [
>         "enclosure.mouth.viseme",
>         "enclosure.mouth.display"
>       ],
>       "enable_old_audioservice": true,
>       "debug": false,
>       "log_level": "INFO",
>       "play_wav_cmdline": "pw-play %1",
>       "play_mp3_cmdline": "pw-play %1",
>       "first_boot": false
>     }
>   },
>   "tts": {
>     "ovos-tts-plugin-server": {
>       "host": [
>         "https://tts.smartgic.io/matxa",
>         "https://matxa.ziggyai.online"
>       ],
>       "voice": "central/grau"
>     },
>     "module": "ovos-tts-plugin-server"
>   },
>   "stt": {
>     "ovos-stt-plugin-server": {
>       "url": [
>         "https://stt.smartgic.io/citrinet",
>         "https://citrinetstt.ziggyai.online"
>       ]
>     },
>     "module": "ovos-stt-plugin-server",
>     "fallback_module": ""
>   },
>   "system_unit": "metric",
>   "temperature_unit": "celsius",
>   "windspeed_unit": "km/h",
>   "precipitation_unit": "mm",
>   "time_format": "full",
>   "spoken_time_format": "half",
>   "date_format": "DMY"
> }
> ```

I removed the `ovos-PHAL-plugin-wifi-setup` package which is the culprit.

**Comment by suvanbanerjee** [comment]
thanks @goldyfruit for the pr's it made my job super easy 😅, i think this pr is ready to review and push, if there are any more changes please require do tell me

**Comment by goldyfruit** [comment]
> thanks @goldyfruit for the pr's it made my job super easy 😅

I mostly did it as you can learn from it for the next time.

**Comment by goldyfruit** [comment]
@suvanbanerjee have you tried this PR?

**Pull request merged**